pallets/dapp-staking/src/lib.rs

```diff
@@ -1617,6 +1603,19 @@ pub mod pallet {
             Self::update_ledger(&account, ledger)?;
             StakerInfo::<T>::remove(&account, &smart_contract);
 
+            // Return the `StakeAmount` that has max total value.
+            let mut unstake_amount = unstake_amount_iter
+                .iter()
+                .max_by(|a, b| a.total().cmp(&b.total()))
+                // At least one value exists, otherwise we wouldn't be here.
+                .ok_or(Error::<T>::InternalUnstakeError)?
+                .clone();
+
+            // Ensure we use the current era instead of potentially next era
+            unstake_amount.era = current_era;
+
+            assert_eq!(unstake_amount.total(), amount);
```

Comment:
This check is irrelevant in `inner_unstake` ?

Reply:
To add to this, since I didn't write it yesterday - please remove this from the `lib.rs`.

Reply:
It's something we used to have - [reference](https://github.com/AstarNetwork/Astar/blob/4edc57ecf84fd194526bed0657ea195bb2f36d70/pallets/dapp-staking/src/lib.rs#L1606) 😅
But I agree, it's gone now!

Reply:
Not sure how it ended up there, but yeah, it shouldn't be used in production code.